cancat/utils/types.py

```diff
@@ -0,0 +1,89 @@
+# Various types used by CanCat utility functions.
+
+import re
+
+# Test if this is python2 or python3
+try:
+    _ = xrange(1, 2)
+    _range_func = xrange
+except NameError:
+    _range_func = range
+
+def _dec_range(val, increment=1):
+    parts = re.split(r'-', val)
+    if len(parts) == 1:
+        return int(parts[0])
+    else:
+        return _range_func(int(parts[0]), int(parts[1]) + 1, increment)
+
+def _hex_range(val, increment=1):
+    parts = re.split(r'-', val)
+    if len(parts) == 1:
+        return int(parts[0], 16)
+    else:
+        return _range_func(int(parts[0], 16), int(parts[1], 16) + 1, increment)
+
+
+class SparseRange(tuple):
+    def __new__(cls, val, increment=1):
+        return super(SparseRange, cls).__new__(cls, (_dec_range(v, increment) for v in val.split(',')))
```

Comment:
This is a utility type created for the command line arguments to canmap.  The goal is to make it easier for command line parsing to pass multiple ranges of values.  So if you want to scan for DIDs 0x1000-0x1100 and 0xF100-0xF200 you can specify `1000-1100,f100-f200` instead of having to include the entire range you are interested in.